iris_grib/_load_convert.py

```diff
@@ -1343,21 +1355,39 @@ def hybrid_factories(section, metadata):
             coord = DimCoord(scaledValue, standard_name='model_level_number',
                              attributes=dict(positive='up'))
             metadata['aux_coords_and_dims'].append((coord, None))
+
+            if typeOfFirstFixedSurface == 118:
+                # height
+                level_value_name = 'level_height'
+                level_value_units = 'm'
+                factory_class = HybridHeightFactory
+                factory_args = [{'long_name': level_value_name},
+                                {'long_name': 'sigma'},
+                                Reference('orography')]
+            else:
+                # pressure
+                level_value_name = 'level_pressure'
+                level_value_units = 'Pa'
+                factory_class = HybridPressureFactory
+                factory_args = [{'long_name': 'level_pressure'},
+                                {'long_name': 'sigma'},
+                                Reference('surface_air_pressure')]
+
             # Create the level pressure scalar coordinate.
             pv = section['pv']
             offset = scaledValue
-            coord = DimCoord(pv[offset], long_name='level_pressure',
-                             units='Pa')
+            coord = DimCoord(pv[offset], long_name=level_value_name,
+                             units=level_value_units)
             metadata['aux_coords_and_dims'].append((coord, None))
             # Create the sigma scalar coordinate.
             offset += NV // 2
```

Comment:
Probably clearer/safer to replace the "+=" with a standalone calculation, i.e. `= scaledValue + NV // 2`